From 06ffa38690908f0c8bc807b5333b60a0b5100666 Mon Sep 17 00:00:00 2001
From: Zhong Hongbo <hongbo.zhong@windriver.com>
Date: Fri, 26 Apr 2013 17:12:00 +0800
Subject: [PATCH 09/13] arm: zynq: Fix sparse errors in platsmp.c

git://git.xilinx.com/linux-xlnx.git master
commit 23767eb028bf9188a6ab98cdd64cf88d8bf52e09

Warning log:
arch/arm/mach-zynq/platsmp.c:60:3: warning: incorrect type in argument 1 (different address spaces)
arch/arm/mach-zynq/platsmp.c:60:3:    expected void const volatile [noderef] <asn:2>*<noident>
arch/arm/mach-zynq/platsmp.c:60:3:    got unsigned char [usertype] *
...

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 arch/arm/mach-zynq/platsmp.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-zynq/platsmp.c b/arch/arm/mach-zynq/platsmp.c
index 1bf4724..a6b8cff 100644
--- a/arch/arm/mach-zynq/platsmp.c
+++ b/arch/arm/mach-zynq/platsmp.c
@@ -51,7 +51,7 @@ void __cpuinit platform_secondary_init(unsigned int cpu)
 }
 
 /* Store pointer to ioremap area which points to address 0x0 */
-static u8 *zero;
+static u8 __iomem *zero;
 
 /*
  * Store pointer to SLCR registers. SLCR driver can't be used because
@@ -59,7 +59,7 @@ static u8 *zero;
  */
 #define SLCR_UNLOCK	0xDF0D
 
-static u8 *slcr;
+static u8 __iomem *slcr;
 
 int zynq_cpu1_start(u32 address)
 {
@@ -174,7 +174,7 @@ void __init platform_smp_prepare_cpus(unsigned int max_cpus)
 	}
 #else
 	/* The first three addresses at zero are already mapped */
-	zero = (u8 *)CONFIG_PAGE_OFFSET;
+	zero = (__force u8 __iomem *)CONFIG_PAGE_OFFSET;
 #endif
 
 	/*
-- 
1.7.5.4

